components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -102,6 +103,8 @@ handler.on(WalletPageActions.restoreWallet.getType(), async (store: Store, paylo
   store.dispatch(WalletPageActions.setShowIsRestoring(false))
   if (payload?.completeWalletSetup) {
     store.dispatch(WalletPageActions.walletSetupComplete(payload.completeWalletSetup))
+  } else {
+    store.dispatch(WalletPageActions.reportOnboardingAction(OnboardingAction.RESTORED_WALLET))
```

Comment:
No need to dispatch an extra action here. You can call `getWalletPageApiProxy().braveWalletP3A.reportOnboardingAction` directly here